Add new endpoint - setHRInterview

AnnKirienko opened the pull request.

AnnKirienko committed:
add new dto for marks from interview

also add services and some fields in UnitOfWork - they dont work correctly

AnnKirienko committed:
delete useless comment, update skillKnowledgeService

AnnKirienko committed:
update post method

 it needs to write some new things

AnnKirienko committed:
Merge branch 'main' into Controller

AnnKirienko committed:
add new mapping profile, remake set interview results in HR Controller

AnnKirienko committed:
delete useless map profile

AnnKirienko committed:
deleted useless things, remade skillKnowledgeService

AnnKirienko committed:
update mapping

AnnKirienko committed:
update2.0

AnnKirienko committed:
add method save with return id in BaseModel

AnnKirienko committed:
add map

AnnKirienko committed:
trying to fix

AnnKirienko committed:
add working Post HRInterview Method

delete unnecessary things, add logic

AnnKirienko committed:
Merge branch 'main' into Controller

AnnKirienko committed:
add new things for setWithDecline status

DShcharbakou commented on BLL/DTO/HRInterviewDTOWithDecline.cs:
```diff
@@ -8,6 +8,6 @@ namespace BLL.DTO
 {
   public  class HRInterviewDTOWithDecline : HRInterviewDTO
```
would be better call this method like HRInterviewDTOWithStatus

DShcharbakou commented on BLL/Interfaces/ISkillKnowledgeService.cs:
```diff
@@ -0,0 +1,11 @@
+{
+    public interface ISkillKnowledgeService
+    {
+      //  IEnumerable<SkillKnowledgeDTO> GetList();
```
delete this

DShcharbakou commented on BLL/MappingProfiles/MappingProfile.cs:
```diff
@@ -15,8 +15,8 @@ public class MappingProfile : Profile
             CreateMap<Candidate, CandidateDTO>();
-              /*  .ForMember(cd => cd.ID,
-                 cd => cd.MapFrom(cd => cd.Id);*/
+            /*  .ForMember(cd => cd.ID,
+               cd => cd.MapFrom(cd => cd.Id);*/
```
delete this

DShcharbakou commented on BLL/Services/InterviewService.cs:
```diff
@@ -42,11 +43,22 @@ public IEnumerable<InterviewDTO> GetList()
+        {
+            var interview = _mapper.Map<Interview>(hrInterviewDTOWithDecline);
+            _db.Interviews.Save(interview);
+            //var candSand = _db.Interviews.FindWithSpecificationPattern(new InterviewStatusSpecification()).FirstOrDefault(x => x.CandidateID == interview.CandidateID);
+            //var cand = candSand.Candidate.CandidateSandboxes.Where(x => x.Status.Id == hrInterviewDTOWithDecline.StatusID);
+            //hrInterviewDTOWithDecline.StatusID
```
delete this

DShcharbakou commented on BLL/Services/SkillKnowledgeService.cs:
```diff
@@ -0,0 +1,52 @@
+        }
+        public void AddSkillKnowledge(IEnumerable<SkillKnowledgeDTO> skillKnowledgeDtoList)
+        {
+            foreach (var skillKnowledgeDto in skillKnowledgeDtoList)
```
bad practice. Save into database in the loop

AnnKirienko replied:
we can remake this in future

DShcharbakou commented on BLL/Services/SkillKnowledgeService.cs:
```diff
@@ -0,0 +1,52 @@
+        }
+
+
+        /* 
+                public IEnumerable<SkillKnowledgeDTO> GetList()
+                {
+                    return _mapper.Map<IEnumerable<SkillKnowledge>, List<SkillKnowledgeDTO>>(_db.SkillKnowledges.GetAll());
+                }     
+              */
```
delete this

DShcharbakou commented on UI/Controllers/HRController.cs:
```diff
@@ -49,17 +52,29 @@ public CandidateDTO Get(int id)
+
+            var skillKnowledgeDTOList = _mapper.Map<IEnumerable<SkillKnowledgeDTO>>(hRInterviewWithDecline);
+            _skillKnowledgeService.AddSkillKnowledge(skillKnowledgeDTOList);
+        }
```
I don't see differences between InterviewResults and InterviewResultsWithDeclineStatus

AnnKirienko replied:
I am too, but I don't know how I can use endpoint method

AnnKirienko committed:
update SetInterviewWithStatus

add new method to the sandbox service
but SetInterviewWithStatus at this moment doesn't work

AnnKirienko committed:
Merge branch 'main' into Controller

AnnKirienko committed:
new updates for save candidateSandbox

(unfortunately, not work)

AnnKirienko committed:
Merge branch 'main' into Controller

AnnKirienko committed:
fix merge problems

AnnKirienko committed:
delete comments

AnnKirienko committed:
updated and finished not worked methods

SetHRInterviewWithStatus is work, RegistrationCandidate is also work

AnnKirienko committed:
add new endpoint - set status
